graphql-code-generator/src/main/java/io/ballerina/graphql/generator/utils/BallerinaFileMerger.java

```diff
@@ -0,0 +1,260 @@
+                mergedMembers.add(existingMember);
+            }
+        }
+        
+        // Add any remaining generated members (newly added types/functions)
+        for (ModuleMemberDeclarationNode member : generatedMemberMap.values()) {
+            if (member != null) {
+                mergedMembers.add(member);
+            }
+        }
+        
+        return NodeFactory.createNodeList(mergedMembers);
+    }
+
+    /**
+     * Creates a map of module members keyed by their signature.
+     *
+     * @param members the module members
+     * @return a map of members keyed by signature
+     */
+    private static Map<String, ModuleMemberDeclarationNode> createMemberMap(
+            NodeList<ModuleMemberDeclarationNode> members) {
+        if (members == null) {
+            return new HashMap<>();
+        }
+        
+        Map<String, ModuleMemberDeclarationNode> memberMap = new HashMap<>();
+        
+        for (ModuleMemberDeclarationNode member : members) {
+            if (member != null) {
+                String key = getMemberKey(member);
+                if (key != null && !key.isEmpty()) {
+                    memberMap.put(key, member);
+                }
+            }
+        }
+        
+        return memberMap;
+    }
+
+    /**
+     * Gets a unique key for a module member.
+     *
+     * @param member the module member
+     * @return a unique key for the member
+     */
+    private static String getMemberKey(ModuleMemberDeclarationNode member) {
+        if (member == null) {
+            return null;
+        }
+        
+        if (member.kind() == SyntaxKind.FUNCTION_DEFINITION) {
+            FunctionDefinitionNode function = (FunctionDefinitionNode) member;
+            if (function.functionName() != null) {
+                return "function:" + function.functionName().text();
+            }
+        } else if (member.kind() == SyntaxKind.SERVICE_DECLARATION) {
+            ServiceDeclarationNode service = (ServiceDeclarationNode) member;
+            if (service.absoluteResourcePath() != null && service.absoluteResourcePath().size() > 0) {
+                return "service:" + service.absoluteResourcePath().get(0).toString();
+            }
+            return "service";
+        } else if (member.kind() == SyntaxKind.TYPE_DEFINITION) {
+            TypeDefinitionNode type = (TypeDefinitionNode) member;
+            if (type.typeName() != null) {
+                return "type:" + type.typeName().text();
+            }
+        } else if (member.kind() == SyntaxKind.MODULE_VAR_DECL) {
+            ModuleVariableDeclarationNode var = (ModuleVariableDeclarationNode) member;
+            // For variables, we'll use a simple approach
+            return "variable:" + var.toString().hashCode();
+        }
+        
+        // For other types, use a hash of the toString representation
+        return member.kind().name() + ":" + member.toString().hashCode();
+    }
+}
```

Comment:
Add an empty line at the end of each file.

```suggestion
}

```